packages/venia-concept/src/components/Checkout/paymentsForm.js

```diff
@@ -1,90 +1,230 @@
-import { Form } from 'informed';
+import React, { Component, Fragment } from 'react';
+import { Form, Text } from 'informed';
 import { array, bool, func, shape, string } from 'prop-types';
 
-import Button from 'src/components/Button';
+import BraintreeDropin from './braintreeDropin';
 import Label from './label';
-import Select from 'src/components/Select';
+import Button from 'src/components/Button';
+import Checkbox from 'src/components/Checkbox';
 
 import classify from 'src/classify';
 import defaultClasses from './paymentsForm.css';
 
+import isObjectEmpty from 'src/util/isObjectEmpty';
+
+const DEFAULT_FORM_VALUES = {
+    addresses_same: true
+};
+
 class PaymentsForm extends Component {
     static propTypes = {
-        availablePaymentMethods: array.isRequired,
         cancel: func.isRequired,
         classes: shape({
+            address_check: string,
             body: string,
+            button: string,
+            braintree: string,
+            city: string,
             footer: string,
             heading: string,
-            paymentMethod: string
+            postcode: string,
+            region_code: string,
+            street0: string,
+            textInput: string
+        }),
+        initialValues: shape({
+            addresses_same: bool,
+            city: string,
+            postcode: string,
+            region_code: string,
+            street0: array
         }),
-        paymentMethod: string,
         submit: func.isRequired,
         submitting: bool
     };
 
-    static defaultProps = {
-        availablePaymentMethods: [{}]
+    state = {
+        isRequestingPaymentNonce: false
     };
 
     render() {
-        const {
-            availablePaymentMethods,
-            classes,
-            paymentMethod,
-            submitting
-        } = this.props;
-
-        const selectablePaymentMethods = availablePaymentMethods.map(
-            ({ code, title }) => ({ label: title, value: code })
+        const { classes, initialValues } = this.props;
+        const { formChildren } = this;
+
+        let initialFormValues;
+        if (isObjectEmpty(initialValues)) {
+            initialFormValues = DEFAULT_FORM_VALUES;
+        }
+        // We have some initial values, use them.
+        else {
+            if (initialValues.sameAsShippingAddress) {
+                // If the addresses are the same, don't populate any fields
+                // other than the checkbox with an initial value.
+                initialFormValues = {
+                    addresses_same: true
+                };
+            } else {
+                // The addresses are not the same, populate the other fields.
+                initialFormValues = {
+                    ...initialValues,
+                    // Note: we have to 'rename' this property - it doesn't match
+                    // the form field's name ("addresses_same").
+                    addresses_same: initialValues.sameAsShippingAddress
```

Comment:
```suggestion
                    addresses_same: false
```

Would be identical, no?

Reply:
Fixed in https://github.com/magento-research/pwa-studio/pull/744/commits/edbf60386b5a24d8e483365a477255e531ae8e81.